### DISCUSSION TITLE
Need help running the program

### DISCUSSION THREAD

**Discussion by erikvalen** [ReagentX/imessage-exporter#244] [Q&A] [answered]
Hello,

First off, thank you for building this tool. After much trial and error I've managed to install the program (I believe at least haha.)
I've ran the command "cargo install imessage-exporter" and completed it. I've then navigated to the /bin/ folder as requested and tried running any of the aforementioned commands on the how to use page (Example: % imessage-exporter -f html -c compatible). No luck, I just keep getting the message "fg: no current job"

Any help would be greatly appreciated!

Thanks

**Comment by ReagentX** [comment, marked as answer]
In your example `% imessage-exporter -f html -c compatible`, are you including the `%` as part of your command?

In the examples in the readme, the `%`s are intended to serve as the primary prompt character `PS1`, after which you would enter the command `imessage-exporter -f html -c compatible`.

If you entered the `%` as part of the command, it is probably interpreting it for job control, as detailed [here](https://superuser.com/questions/826562/percent-sign-in-front-of-shell-command).

**Reply by erikvalen** [reply]
So when I enter the command `imessage-exporter -f html -c compatible` it just says `zsh: command not found: imessage-exporter`

I have used the command cd and navigated to the /bin/ folder as directed. Am I doing something wrong? Sorry, I have never used a mac before, let alone mac terminal :(

**Reply by ReagentX** [reply]
This sounds like you didn't install the software. I'm not sure what you mean about `/bin/`, you shouldn't have to mess with that stuff.

**Reply by erikvalen** [reply]
![install](https://github.com/ReagentX/imessage-exporter/assets/166887523/490fd4b4-c4ed-4b85-9add-66946d285ee5)
Here's the terminal after installing. You can see at the bottom that it says to add `/users/usernamehere/.cargo/bin` to the PATH. I figured that meant to use cd to change the directory to that address. I tried running the commands directly after install in the same terminal without any effect as well.

When I go to the directory mentioned, and run the imessage-exporter file it opens a new terminal window with command instructions but no ability to enter a command.

**Reply by ReagentX** [reply]
Have you restarted your terminal after installing `cargo`? If not, that will update some things behind-the-scenes that will probably resolve this. 

If you have done that already and you still get the issue, it means that `cargo`'s install directory isn't present on your `PATH`. It's not something you would solve with `cd`, the `PATH` is an environment variable that contains a list of folders that contain programs you can run in your shell. What this error is telling you is that the path `/Users/username/.cargo/bin` is not included in that list, so your shell doesn't know about the programs it contains.

You can do 2 things:

1) if you `cd` to that `/bin` directory, you can run `imessage-exporter` by prefixing the command with `./`, which tells your shell to look in the current directory and not use your `PATH`. In that case, your command would be: `./imessage-exporter -f html -c compatible`
2) run the command `export PATH="$HOME/.cargo/bin:$PATH"` first (to update your `PATH`) and then run your original command (without prefixing with `./`). 

The reason when you double-click the program it emits the instructions is that double-clicking an app like that is the same as running it without any arguments, and I set that default invocation (I.e. just entering `imessage-exporter`) to emit the help text. I don't know why Apple prevents the use of the shell after you do this, but I assume they have their reasons.

**Reply by erikvalen** [reply]
Finally got it working! I believe it was the PATH that was the issue! Thank you so much for this tool and all the help!
